gtsfm/frontend/global_descriptor/netvlad_global_descriptor.py

```diff
@@ -0,0 +1,45 @@
+"""Wrapper around the NetVLAD global image descriptor.
+
+Based on Arandjelovic16cvpr:
+"NetVLAD: CNN architecture for weakly supervised place recognition"
+https://arxiv.org/pdf/1511.07247.pdf
+
+NetVLAD, is a new generalized VLAD layer, inspired by the “Vector of Locally Aggregated Descriptors”
+image representation commonly used in image retrieval
+
+Whereas bag-of-visual-words aggregation keeps counts of visual words, VLAD stores the sum of residuals
+(difference vector between the descriptor and its corresponding cluster centre) for each visual word.
+
+Authors: John Lambert
+"""
+
+import numpy as np
+import torch
+from torch import nn
+
+from gtsfm.common.image import Image
+from gtsfm.frontend.global_descriptor.global_descriptor_base import GlobalDescriptorBase
+from thirdparty.hloc.netvlad import NetVLAD
+
+
+class NetVLADGlobalDescriptor(GlobalDescriptorBase):
```

Comment:
no documentation?

Reply:
Just a short description is fine to avoid mypy warnings.

Reply:
unresolved